### ISSUE TITLE
Days of Stock should not count days the depot is not operating

### ISSUE THREAD

**Issue by jmcameron** [third-culture-software/bhima#4876]
When we compute CMM using the "Days in Stock", we need to make sure that "Days in stock" is the number of days that the depot was open/operating AND there was stock.   

Having stock or not during days that there is no chance to dispense/transfer stock is irrelevant to CMM.  That implies that the 30.5 number we see in the CMM formulas should probably be replaced by the maximum number days the facility is open during the period in question.  Note that this number (max days in stock) depends on the depot and will vary from month to month (depending on weekends hit, holidays, closures, etc).

_Originally posted by @jmcameron in https://github.com/IMA-WorldHealth/bhima/pull/4866#issuecomment-682802049_

**Comment by jeremielodi** [comment]
Is there any place where we specify the period when the depot is not operating?

**Comment by jmcameron** [comment]
> Is there any place where we specify the period when the depot is not operating?

Not now.   And the default CMM calculations that we have received use the 30.5 number which is a clear indication that they are assuming "average months" which also implies that they are not factoring in any "off" days.

I was imagining that (eventually) each depot could indicate what schedule they normally work (5 days/week, 6 days/week, 24/7) as the basis for this number (this is probably all the accuracy that is needed).   However, if there are a lot of national holidays, that could also be taken into account.

However when subdepots (without the Bhima software) are reporting their  aggregate consumption numbers upstream to the managing Depot (using Bhima software), they would report something like:  For the following date rate, we consumed N of stock item X and operated M many days.

**Comment by jniles** [comment]
This is a good idea, and technically correct.  But I don't think I've seen it before in any document reported in the national health statistics.  It may just be that the number of days depots are closed is so small that they shouldn't affect any calculations.

**Comment by jmcameron** [comment]
Aren't most depots closed at least one or two days a week?  That is 14% or 28% effectively.  That is probably not "small" when it comes to estimating CMMs (I think!)

**Comment by jniles** [comment]
I don't think that health centers are ever closed.  They are the primary care centers.  That is part of the reason I asked the team to look into this on our chat.

I know that the IMCK administration doesn't work on Sundays, but I think (hope!) the pharmacy is open.   IMCK is the only institution I know of that doesn't have a skeleton admin team to support the needs of the hospital on Sundays.  Kintambo, Karawa, and Vanga all have 24 hour service.  Certain services (radiology, surgery, etc) aren't officially open, but the lab and pharmacy are.
